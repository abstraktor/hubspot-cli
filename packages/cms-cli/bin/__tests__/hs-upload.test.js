/**
 * @jest-environment ./CommandTestEnvironment.js
 */

// const path = require('path');
const { node: execNode } = require('execa');

const {
  hsPath,
  hscmsPath,
  // remoteRoot,
  // localTestDir
} = global;

console.log(hsPath);

const testEach = test.each`
  exe        | script
  ${'hs'}    | ${hsPath}
  ${'hscms'} | ${hscmsPath}
`;

testEach('$exe upload --help', async ({ script }) => {
  const { stdout } = await execNode(script, ['upload', '--help']);
  expect(stdout).toMatchSnapshot();
});

// testEach('$exe upload (missing src)', async ({ script }) => {
//   let stderr;
//   try {
//     await execNode(script, ['upload']);
//   } catch (err) {
//     ({ stderr } = err);
//   }
//   expect(stderr).toBeTruthy();
// });

// testEach('$exe upload --portal=invalid', async ({ script }) => {
//   let stderr;
//   try {
//     await execNode(script, ['upload', '--portal=invalid']);
//   } catch (err) {
//     ({ stderr } = err);
//   }
//   expect(stderr).toBeTruthy();
// });

// testEach(`$exe upload ${remoteRoot}/js`, async ({ script }) => {
//   const src = `${remoteRoot}/js`;
//   const dest = path.join(localTestDir, 'foo');
//   const { stdout } = await execNode(script, ['upload', src, dest]);
//   expect(stdout).toBeTruthy();
// });

// testEach('$exe upload /does-not-exist', async ({ script }) => {
//   const src = '/does-not-exist';
//   const dest = localTestDir;
//   const { stderr } = await execNode(script, ['upload', src, dest]);
//   expect(stderr).toBeTruthy();
// });
