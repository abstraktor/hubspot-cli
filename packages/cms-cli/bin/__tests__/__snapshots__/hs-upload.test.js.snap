// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hs upload --help 1`] = `
"Usage: hs upload [options] <src> <dest>

Upload a folder or file from your computer to the HubSpot CMS

Options:
  -V, --version      output the version number
  --config <config>  path to a config file
  --portal <portal>  HubSpot portal id or name from config
  --debug            set log level to debug
  --mode <mode>      write to <draft | publish>
  -h, --help         output usage information"
`;

exports[`hscms upload --help 1`] = `
"Usage: hscms upload [options] <src> <dest>

Upload a folder or file from your computer to the HubSpot CMS

Options:
  -V, --version      output the version number
  --config <config>  path to a config file
  --portal <portal>  HubSpot portal id or name from config
  --debug            set log level to debug
  --mode <mode>      write to <draft | publish>
  -h, --help         output usage information"
`;
